bn-apps/memberships-management/membership-service/src/main/kotlin/net/corda/businessnetworks/membership/bno/support/BusinessNetworkAwareInitiatedFlow.kt

```diff
@@ -0,0 +1,32 @@
+package net.corda.businessnetworks.membership.bno.support
+
+import co.paralleluniverse.fibers.Suspendable
+import net.corda.businessnetworks.membership.bno.service.DatabaseService
+import net.corda.businessnetworks.membership.common.CounterPartyMembershipNotActive
+import net.corda.businessnetworks.membership.common.CounterPartyNotAMemberException
+import net.corda.core.flows.FlowSession
+import net.corda.core.identity.Party
+
+abstract class BusinessNetworkAwareInitiatedFlow<out T>(val flowSession: FlowSession) : BusinessNetworkAwareFlow<T>() {
+
+    @Suspendable
+    override fun call(): T {
+        confirmInitiatorIsAMemberOfOurBN(flowSession.counterparty)
+        return afterOtherPartyMembershipChecked()
+    }
+
+    @Suspendable
+    abstract fun afterOtherPartyMembershipChecked() : T
+
+    @Suspendable
+    private fun confirmInitiatorIsAMemberOfOurBN(initiator : Party) {
```

Comment:
maybe just `verifyMembership()`? As it's taking a party as a parameter anyway

Reply:
ok